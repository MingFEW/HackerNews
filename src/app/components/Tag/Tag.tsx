import React from 'react'
import tw from 'twin.macro'

import { capitalizeFirstLetter } from '@/utils/capitalizeFirstLetter'

interface TagProps {
  title: string
}

export const Tag: React.FC<TagProps> = props => {
  const { title, ...rest } = props
  return (
    <div
      css={[
        tw`flex items-center justify-center h-8 px-3 rounded-2xl`,
        tw`bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500`
      ]}
      {...rest}
    >
      <div tw="text-xs sm:text-sm text-white font-bold mb-[2px]">{capitalizeFirstLetter(title)}</div>
    </div>
  )
}
