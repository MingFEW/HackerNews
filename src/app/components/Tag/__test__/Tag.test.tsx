import { createRenderer } from 'react-test-renderer/shallow'

import { Tag } from '../Tag'

const shallowRenderer = createRenderer()

describe('<Tag />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<Tag title="story" />)
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
