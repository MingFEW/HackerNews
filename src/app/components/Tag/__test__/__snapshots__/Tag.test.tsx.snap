// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render and match the snapshot 1`] = `
<div
  tw="flex items-center justify-center h-8 px-3 rounded-2xl bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500"
>
  <div
    tw="text-xs sm:text-sm text-white font-bold mb-[2px]"
  >
    Story
  </div>
</div>
`;
